src/analysis/retail/paladin/holy/modules/beacons/BeaconAnalyzer.tsx

```diff
@@ -0,0 +1,23 @@
+import TALENTS from 'common/TALENTS/paladin';
+import Analyzer, { Options } from 'parser/core/Analyzer';
+import { BeaconType, BEACON_TYPES } from '../../constants';
+
+class BeaconAnalyzer extends Analyzer {
+  beaconType: BeaconType;
+  beaconBuffIds: readonly number[];
+
+  constructor(options: Options) {
+    super(options);
+
+    if (this.selectedCombatant.hasTalent(TALENTS.BEACON_OF_FAITH_TALENT)) {
+      this.beaconType = 'BEACON_OF_FATH';
+    } else if (this.selectedCombatant.hasTalent(TALENTS.BEACON_OF_VIRTUE_TALENT)) {
+      this.beaconType = 'BEACON_OF_VIRTUE';
+    } else {
+      this.beaconType = 'BEACON_OF_LIGHT';
+    }
+    this.beaconBuffIds = BEACON_TYPES[this.beaconType];
+  }
+}
```

Comment:
If you need to access beaconType in other files, you should export const variables that store `BEACON_OF_<type>` that way other files don't have to compare to a string and can use the const variable instead.

Reply:
If blizzard adds another beacon type, wouldn't we want to have the uses that need to be exhaustive flagged automatically by the expanded enum?

I do think it is cleaner to use an enum for this regardless. Autocompletion is much more reliable for enum types than string unions

Reply:
the autocompletion thing is more about poor editor support than any fundamental issue with typed string unions. Typed string unions have some other issues with inference (ask me about `as const` on the vega specs sometime :skull:), but otherwise the same as enums. I'm aware.